core/commands/keystore.go

```diff
@@ -163,13 +164,24 @@ path can be specified with '--output=<path>' or '-o=<path>'.
 			return err
 		}
 
-		r, err := fsrepo.Open(cfgRoot)
+		// Check repo version, and error out if not matching
+		ver, err := migrations.RepoVersion(cfgRoot)
+		if err != nil {
+			return err
+		}
+		if ver > fsrepo.RepoVersion {
```

Comment:
Hm. I think I got this wrong, actually. We probably need to check for an _exact_ match. I know we've changed the on-disk format before.